packages/types/index.ts

```diff
@@ -18,20 +18,27 @@ export interface Rule<Context = void, RuleOptions = BaseRuleOptions> {
 }
 
 export interface RuleResult {
-  diagnostics: Array<Diagnostic>
+  diagnostics: Array<PartialDiagnostic>
 }
 
-export interface Diagnostic {
+export interface PartialDiagnostic {
   message: string
   fixes?: Array<Fix>
   location: {
     /** Absolute path to a folder or a file that contains the issue. */
     path: string
+    type: 'file' | 'folder'
```

Comment:
issue: this addition is kind of a bummer, I wouldn't want to hurt the plugin authors' experience for that change in the matching logic. Besides, I don't think this is necessary, since the code checks if the glob ends with `**`, we know that this cannot be a file, it has to be a folder.

suggestion: let's roll this one back

Reply:
As I said, I thought it would be useful in the future for other functionalities, e.g.:
- for pretty-printer to print some icons of a file/folder for clarity and faster visual perception of the report
- to filter what diagnostics to display in IDEs (obviously, in an IDE, we can highlight errors only inside files)
- other sorting, rearranging, and partitioning operations on diagnostics
- etc.

And I thought it would be better to initially define which FS node a diagnostic applies to, rather than having to determine it in many places where diagnostics are processed.

So, are you sure we won't need that field in the future? Because if we see many applications for that field in the future, isn't it better to let plugin creators get used to it sooner than introduce a breaking change in the future?